mmtrack/models/track_heads/stark_head.py

```diff
@@ -0,0 +1,472 @@
+                      (feats_flatten_len, bs, embed_dims).
+        """
+        _, bs, _ = x.shape
+        query_embed = query_embed.unsqueeze(1).repeat(
+            1, bs, 1)  # [num_query, embed_dims] -> [num_query, bs, embed_dims]
+
+        enc_mem = self.encoder(
+            query=x,
+            key=None,
+            value=None,
+            query_pos=pos_embed,
+            query_key_padding_mask=mask)
+        target = torch.zeros_like(query_embed)
+        # out_dec: [num_dec_layers, num_query, bs, embed_dims]
+        out_dec = self.decoder(
+            query=target,
+            key=enc_mem,
+            value=enc_mem,
+            key_pos=pos_embed,
+            query_pos=query_embed,
+            key_padding_mask=mask)
+        out_dec = out_dec.transpose(1, 2)
+        return out_dec, enc_mem
+
+
+@HEADS.register_module()
+class StarkHead(BaseModule):
+    """STARK head module for bounding box regression and prediction of
+    confidence score of tracking bbox.
+
+    This module is proposed in
+    "Learning Spatio-Temporal Transformer for Visual Tracking".
+    `STARK <https://arxiv.org/abs/2103.17154>`_.
+
+    Args:
+        num_query (int): Number of query in transformer.
+        transformer (obj:`mmcv.ConfigDict`|dict): Config for transformer.
+            Default: None.
+        positional_encoding (obj:`mmcv.ConfigDict`|dict):
+            Config for position encoding.
+        run_cls_head (bool, optional): whether to run classification head.
+            Defaults to False.
+        run_bbox_head (bool, optional): whether to run bboxes regression head.
+            Defaults to True.
+        bbox_head (obj:`mmcv.ConfigDict`|dict, optional): Config for bbox head.
+            Defaults to None.
+        cls_head (obj:`mmcv.ConfigDict`|dict, optional): Config for
+            classification head. Defaults to None.
+        loss_cls (obj:`mmcv.ConfigDict`|dict): Config of the
+            classification loss. Default `CrossEntropyLoss`.
+        loss_bbox (obj:`mmcv.ConfigDict`|dict): Config of the bbox
+            regression loss. Default `L1Loss`.
+        loss_iou (obj:`mmcv.ConfigDict`|dict): Config of the bbox
+            regression iou loss. Default `GIoULoss`.
+        tran_cfg (obj:`mmcv.ConfigDict`|dict): Training config of
+            transformer head.
+        test_cfg (obj:`mmcv.ConfigDict`|dict): Testing config of
+            transformer head.
+        init_cfg (dict or list[dict], optional): Initialization config dict.
+            Default: None
+    """
+
+    def __init__(self,
+                 num_query=1,
+                 transformer=None,
+                 positional_encoding=dict(
+                     type='SinePositionalEncoding',
+                     num_feats=128,
+                     normalize=True),
+                 run_cls_head=False,
+                 run_bbox_head=True,
+                 bbox_head=None,
+                 cls_head=None,
```

Comment:
can we use whether `cls_head` is None or not to decide whether `run_cls_head` is True or False